content/help/ops/traffic-management/deploy-guidelines/index.md

```diff
@@ -0,0 +1,205 @@
+call (e.g., `kubectl apply -f myVirtualServiceAndDestinationRule.yaml` is not sufficient because the
+resources propagate (from the configuration server, i.e., Kubernetes API server) to the Pilot instances in an eventually consistent manner. If the
+`VirtualService` using the subsets arrives before the `DestinationRule` where the subsets are defined, the Envoy configuration generated by Pilot would refer to non-existent upstream pools, resulting in HTTP 503 errors until all configuration objects are available to Pilot.
+
+To make sure services will have zero down-time when configuring routes with subsets, follow a "make-before-break" process as described below:
+
+* When adding new subsets:
+
+    1. Update `DestinationRules` to add a new subset first, before updating any `VirtualServices` that use it. Apply the rule using `kubectl` or any platform-specific tooling.
+
+    1. Wait a few seconds for the `DestinationRule` configuration to propagate to the Envoys
+
+    1. Update the `VirtualService` to refer to the newly added subsets.
+
+* When removing subsets:
+
+    1. Update `VirtualServices` to remove any references to a subset, before removing the subset from a `DestinationRule`.
+
+    1. Wait a few seconds for the `VirtualService` configuration to propagate to the Envoys
+
+    1. Update the `DestinationRule` to remove the unused subsets
+
+## Route rules have no effect on ingress gateway requests
+
+Let's assume you are using an ingress `Gateway` and corresponding `VirtualSerive` to access an internal service.
+For example, your `VirtualService` looks something like this:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp
+spec:
+  hosts:
+  - "myapp.com" # or maybe "*" if you are testing without DNS using the ingress-gateway IP (e.g., http://1.2.3.4/hello)
+  gateways:
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /hello
+    route:
+    - destination:
+        host: helloworld.default.svc.cluster.local
+  - match:
+    ...
+{{< /text >}}
+
+You also have a `VirtualService` which routes traffic for the helloworld service to a particular subset:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: helloworld
+spec:
+  hosts:
+  - helloworld.default.svc.cluster.local
+  http:
+  - route:
+    - destination:
+        host: helloworld.default.svc.cluster.local
+        subset: v1
+{{< /text >}}
+
+In this situation you will notice that requests to the helloworld service via the ingress gateway will
+not be directed to subset v1 but instead will continue to use default round-robin routing.
+
+The ingress requests are using the gateway host (e.g., `myapp.com`)
+which will activate the rules in the myapp `VirtualService` that routes to any endpoint in the helloworld service.
+On the other hand, internal requests with host `helloworld.default.svc.cluster.local` will use the
```

Comment:
Avoid idioms such as "On the other hand"

Reply:
Replace "On the other hand, internal requests with host" with "With internal requests containing the host"